ldm/modules/textual_inversion_manager.py

```diff
@@ -165,13 +167,13 @@ def _inject_tokens_and_assign_embeddings(self, ti: TextualInversion) -> int:
             )
 
         trigger_token_id = self._get_or_create_token_id_and_assign_embedding(
```

Comment:
This way of adding and removing the brackets at the time that triggers are turned into tokens and vice versa does not feel right and is breaking the regression tests as well.

I would suggest that instead of this you revisit the `_parse_embedding_*` methods and have them always return the trigger strings with the angle brackets. The issue you are trying to address is caused by the fact that some embeds use brackets around their trigger strings, and others do not.

I would also suggest that the filename (or directory name in the case of `learned_embeds.bin`) always be used as the trigger, irrespective of the trigger contained within the embed file. The embeddings should only ever be triggered by `<dog>` and not by `dog`.

Reply:
But `learned_embeds.bin can contain multiple tokens, so... rename first?
Also thank to this i spotted bug that we currently loading only last token from v4 embedding:
![image](https://user-images.githubusercontent.com/7768370/230743569-28e27ae2-0119-4fe7-84a6-903ab0a08358.png)

Other points - done

Reply:
Do we have any examples in the wild of embeddings with multiple tokens? We should handle this properly.